source/robot_model/include/robot_model/Model.hpp

```diff
@@ -109,38 +121,48 @@ class Model {
   void init_model();
 
   /**
-   * @brief Compute the jacobian from a given joint state at the joint level given in parameter
+   * @brief Compute the jacobian from a given joint state at the frame in parameter
    * @param joint_state containing the joint values of the robot
-   * @param joint_id id of the joint at which to compute the jacobian
+   * @param joint_id id of the frame at which to compute the jacobian
    * @return the jacobian matrix
    */
-  const StateRepresentation::Jacobian compute_jacobian(const StateRepresentation::JointState& joint_state, unsigned int joint_id);
+  StateRepresentation::Jacobian compute_jacobian(const StateRepresentation::JointState& joint_state, int frame_id);
```

Comment:
keep the `unsigned int` type here I would say

Reply:
Pinocchio uses int

Reply:
ok, in that case `int` all around. I redact my previous comments about uints.

Reply:
just saw this discussion you had after making the same comment. Ok for removing the unsigned in that case although you can try size_t as I think it makes a conversion. However, I would remove the const reference for this type of variable